src/Utils/TypeInfo.php

```diff
@@ -363,52 +363,42 @@ static function ($arg) use ($node) {
                     $inputFieldType = $inputField ? $inputField->getType() : null;
                 }
                 $this->defaultValueStack[] = $inputField && $inputField->defaultValueExists() ? $inputField->defaultValue : Utils::undefined();
-                $this->inputTypeStack[]    = Type::isInputType($inputFieldType) ? $inputFieldType : null;
+                $this->inputTypeStack[]    = $inputFieldType;
                 break;
 
             case $node instanceof EnumValueNode:
                 $enumType  = Type::getNamedType($this->getInputType());
                 $enumValue = null;
                 if ($enumType instanceof EnumType) {
-                    $enumValue = $enumType->getValue($node->value);
+                    $this->enumValue = $enumType->getValue($node->value);
                 }
                 $this->enumValue = $enumValue;
                 break;
         }
     }
 
     /**
-     * @return Type
+     * @return (Type & OutputType) | null
```

Comment:
you don't have spaces elsewhere

```suggestion
     * @return (OutputType&Type)|null
```

Reply:
I'll have to take another look at this, but basically the challenge for all this is that I have to satisfy:

1. phpstan/phpdoc-parser
2. code sniffer
3. scrutinizer

In this case,  scrutinizer's static checker is the problem--if I restore the preferred syntax, then it will falsely report incompatibility with the member type. I'll see about opening a bug report with them...

Reply:
IMO we can ignore scrutinizer and report problem to them

Reply:
Okay. I worked really hard to get Scrutinizer to pass, so let's save this for last and I will report the problem to them first thing after work. If they don't respond and there are no other review comments, I'll change this back.

Reply:
Sure there's no point to force it. I'd rather have a sniff for this in the future anyway.

Reply:
Absolutely. I'll dig deep into this tonight--there are at least two Scrutinizer bugs I'm aware of, and one in code sniffer. I'll follow up on those, then I'll see if there's an appropriate sniff...

Reply:
I created a dummy repo with a minimal reproduction case of this issue:

https://scrutinizer-ci.com/g/shmax/scrutinizer-bugs/inspections/fea47b85-e320-4ec1-aff5-cb978a0a10bc/code-structure/operation/MyClass%3A%3AgetType

Scrutinizer's github repo is no more, but I reported it through their chat-bot, which is evidently the only way to report issues, these days.

Reply:
Still not a peep from Scrutinizer team.

Reply:
I would say we never actually bothered about Scrutinizer anyway. Mostly used this for test coverage reports. So feel free to drop its static analysis in favor of PHPStan.

Reply:
I'm working on it!